model/font_simple.go

```diff
@@ -379,16 +381,27 @@ func (font *pdfFontSimple) ToPdfObject() core.PdfObject {
 	return font.container
 }
 
-// NewPdfFontFromTTFFile loads a TTF font and returns a PdfFont type that can be used in text
-// styling functions.
-// Uses a WinAnsiTextEncoder and loads only character codes 32-255.
+// NewPdfFontFromTTFFile is a wrapper around NewPdfFontFromTTF.
 func NewPdfFontFromTTFFile(filePath string) (*PdfFont, error) {
+	f, err := os.Open(filePath)
+	if err != nil {
+		common.Log.Debug("ERROR: reading TTF font file: %v", err)
+		return nil, err
+	}
+
+	return NewPdfFontFromTTF(f)
+}
+
+// NewPdfFontFromTTF loads a TTF font and returns a PdfFont type that can be
+// used in text styling functions.
+// Uses a WinAnsiTextEncoder and loads only character codes 32-255.
+func NewPdfFontFromTTF(r io.ReadSeeker) (*PdfFont, error) {
 	const minCode = textencoding.CharCode(32)
 	const maxCode = textencoding.CharCode(255)
 
-	ttf, err := fonts.TtfParseFile(filePath)
```

Comment:
As long as the whole font file is loaded into memory anyway, you could read it here instead of reading it at the end, and then feed that to `TtfParseFile`.
```
ttfBytes, err := ioutil.ReadAll(r)
ttf, err := fonts.TtfParse(bytes.NewReader(ttfBytes))
...
stream, err := core.MakeStream(ttfBytes, core.NewFlateEncoder())
```

instead of 
```
ttf, err := fonts.TtfParse(r)
...
r.Seek(0, io.SeekStart)
...
ttfBytes, err := ioutil.ReadAll(r)
...
stream, err := core.MakeStream(ttfBytes, core.NewFlateEncoder())
```

It might be a bit faster as well, because in the second version, both  `fonts.TtfParse()` and  `ioutil.ReadAll()` read from the input reader (which could be a file). In the first version, only `ioutil.ReadAll()` reads from the input reader, then  `fonts.TtfParse()` reads from memory.